swap_meet/vendor.py

```diff
@@ -1,2 +1,73 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        self.inventory = inventory
+        if self.inventory is None:
+            self.inventory = []
+
+    def add(self, item):
+        '''adds item to inventory'''
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        '''removes item from inventory'''
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_category(self, category):
+        '''returns list of items of a given category from inventory'''
+        same_category_items = []
+        for item in self.inventory:
+            if item.category == category:
+                same_category_items.append(item)
+        return same_category_items
+
+    def swap_items(self, vendor, my_item, their_item):
+        '''swaps items between vendor inventories'''
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        else:
+            self.inventory.remove(my_item)
+            vendor.inventory.append(my_item)
+            vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            return True
+    
+    def swap_first_item(self, vendor):
+        '''
+        swaps items between vendor inventories 
+        by first position in inventory
+        '''
+        if self.inventory and vendor.inventory:
+            return self.swap_items(vendor, self.inventory[0], \
+            vendor.inventory[0])
```

Comment:
Love how you re-use your earlier method! One small bug here: according to the README this method should return `False` if either of the inventories is empty, but it actually returns `None`.